pyomo/core/base/indexed_component.py

```diff
@@ -25,22 +25,56 @@
 
 UnindexedComponent_set = set([None])
 
-
 def normalize_index(index):
     """
     Flatten a component index.  If it has length 1, then
     return just the element.  If it has length > 1, then
     return a tuple.
     """
-    ndx = pyutilib.misc.flatten(index)
-    if type(ndx) is list:
-        if len(ndx) == 1:
-            ndx = ndx[0]
+    idx = pyutilib.misc.flatten(index)
+    if type(idx) is list:
+        if len(idx) == 1:
+            idx = idx[0]
         else:
-            ndx = tuple(ndx)
-    return ndx
+            idx = tuple(idx)
+    return idx
 normalize_index.flatten = True
 
+#
+# Get the fully-qualified name for this index.  If there isn't anything
+# in the _data dict (and there shouldn't be), then add something, get
+# the name, and remove it.  This allows us to get the name of something
+# that we haven't added yet without changing the state of the constraint
+# object.
+#
+def _get_indexed_component_data_name(component, index):
```

Comment:
Documentation within this method would be useful.  It's not immediately obvious why you're constructing and then deleting a constraint.